test/helper/launcher/launcher.go

```diff
@@ -0,0 +1,257 @@
+	out, err := l.run("kubectl", "config", "current-context")
+	return err == nil && strings.Contains(out, ExpectedContext)
+}
+
+func (l *Launcher) startKind() error {
+	err := l.silentRun("kind", "create", "cluster")
+	if err == nil {
+		l.clearKind = true
+	}
+	return err
+}
+
+func (l *Launcher) stopKind() error {
+	return l.silentRun("kind", "delete", "cluster")
+}
+
+func (l *Launcher) ensureOperator() error {
+	if !l.isInstalled() {
+		return l.install()
+	}
+	return nil
+}
+
+func (l *Launcher) isInstalled() bool {
+	result, err := l.run("helm", "ls", "-a", "-A")
+	if err != nil || !strings.Contains(result, LauncherTestInstall) {
+		return false
+	}
+	scanner := bufio.NewScanner(strings.NewReader(result))
+	for scanner.Scan() {
+		str := scanner.Text()
+		if strings.Contains(str, LauncherTestInstall) {
+			parts := strings.Split(str, "\t")
+			installedVersion := strings.TrimSpace(parts[len(parts)-1])
+			return l.version == installedVersion
+		}
+	}
+	return false
+}
+
+func (l *Launcher) install() error {
+	err := l.silentRun("helm", "repo", "add", RepoRef, HelmRepoURL)
+	update := false
+	if err != nil {
+		if strings.Contains(err.Error(), "already exists") {
+			update = true
+		} else {
+			return fmt.Errorf("failed to set up mongodb helm chart repo: %w", err)
+		}
+	}
+	if update {
+		if err := l.silentRun("helm", "repo", "update", RepoRef); err != nil {
+			return fmt.Errorf("failed to update mongodb helm chart repo: %w", err)
+		}
+	}
+	if err != nil && !strings.Contains(err.Error(), "already exists") {
+		return fmt.Errorf("failed to set up mongodb helm chart repo: %w", err)
+	}
+	err = l.silentRun("helm", "install", LauncherTestInstall, path.Join(RepoRef, OperatorChart),
+		"--version", l.version, "--atomic", "--set", fmt.Sprintf("atlasURI=%s", AtlasURI))
+	if err == nil {
+		l.clearOperator = true
+	}
+	return err
+}
+
+func (l *Launcher) uninstall() error {
+	return l.silentRun("helm", "uninstall", LauncherTestInstall)
+}
+
+func (l *Launcher) ensureAtlasSecret() error {
+	if !l.atlasSecretAvailable() {
+		l.clearSecret = true
+		return l.kubeCreateAtlasSecret()
+	}
+	return nil
+}
+
+func (l *Launcher) atlasSecretAvailable() bool {
+	return l.silentRun("kubectl", "get", "secret", AtlasSecretName) == nil
+}
+
+func (l *Launcher) kubeCreateAtlasSecret() error {
+	if err := l.silentRun("kubectl", "create", "secret", "generic", AtlasSecretName,
+		fmt.Sprintf("--from-literal=orgId=%s", l.credentials.OrgID),
+		fmt.Sprintf("--from-literal=publicApiKey=%s", l.credentials.PublicKey),
+		fmt.Sprintf("--from-literal=privateApiKey=%s", l.credentials.PrivateKey)); err != nil {
+		return err
+	}
+	return l.silentRun("kubectl", "label", "secret", AtlasSecretName, "atlas.mongodb.com/type=credentials")
+}
+
+func (l *Launcher) kubeDeleteAtlasSecret() error {
+	return l.silentRun("kubectl", "delete", "secret", AtlasSecretName)
+}
+
+func (l *Launcher) kubeApply(yml string) error {
+	return l.pipeRun(yml, "kubectl", "apply", "-f", "-")
```

Comment:
nit: hiding this behind small function masks implementation details of the underlying kubectl command, i.e. it is now known in 1. what namespace `kubeApply` operates in and 2. what command line arguments are used. For clarity my taste would be to inline this otherwise I see the danger of an over-abstraction over kubectl and lack of clarity for the reader what happens here.

Reply:
Why are we arguing about an internal implementation detail? This is not a public API.

The idea is we could replace the launcher with something that does not use the Operator or kubectl directly. The Interface we care about is the one seen from the contract test code.

Reply:
as per comment below, agreed